### PR TITLE
improvement

### PR DESCRIPTION
**Pull request by TheFusion21**
- fixed bug with pydantic validation
 - moved cli params to config file
 - added cli progress bar with tqdm
 - added missing packages to setup.py
 - updated README.md

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -37,13 +37,14 @@
     'numpy',
     'packaging',
     'pillow',
-    'pydantic>=2',
+    'pydantic<=2',
```
Comment (lucidrains) [inline setup.py:40]: i don't think the configs will work with pydantic v1

too many breaking changes

File: imagen_pytorch/cli.py [hunk 2]
```diff
@@ -136,8 +132,23 @@ def train(
     
     # load and add train dataset and valid dataset
     ds = load_dataset(config_data['dataset_name'])
+    
+    train_ds = None
+    
+    # if we have train and valid split we combine them into one dataset to let trainer handle the split
+    if 'train' in ds and 'valid' in ds:
+        train_ds = concatenate_datasets([ds['train'], ds['valid']])
```
Comment (lucidrains) [inline imagen_pytorch/cli.py:140]: nice!

**Comment by lucidrains** [comment]
@TheFusion21 hey Kay! long time

thanks for the PR

only quibble i have is about the pydantic dep version

**Comment by TheFusion21** [comment]
@lucidrains pydantic broke something when loading the cli created checkpoint after your change in this [commit](https://github.com/lucidrains/imagen-pytorch/commit/2527ce3569f1167ab66e406e06b79bbbd3aa9f8e)

can't seem to figure out why

**Comment by lucidrains** [comment]
@TheFusion21 ah, we should migrate everything to pydantic 2

what is the error that you see?

**Comment by lucidrains** [comment]
i'm sure you can figure it out, you're a developer!

**Comment by lucidrains** [comment]
@TheFusion21 if you can get a minimal script that creates the error, i'll sink a few hours and get it fixed

**Comment by TheFusion21** [comment]
breaks here https://github.com/lucidrains/imagen-pytorch/blob/4451ab7025f24113feeab72e73a0be16f2ee9471/imagen_pytorch/configs.py#L68C37-L68C37

can get you a full stack trace later

Will dig into it

**Comment by TheFusion21** [comment]
Hope you had a great holiday/vacation and happy new year ^^

**Comment by lucidrains** [comment]
@TheFusion21 you as well! happy new years!

**Pull request merged**